### ISSUE TITLE
manually specify ffmpeg video encoding options

### ISSUE THREAD

**Issue by mikhailnov** [vkohaupt/vokoscreen#170]
It would be great if Vokoscreen allowed to manually specify ffmpeg encoding options, including the codec. For example, I want to live encode the recording into -c:v libx264 -preset veryslow, I can specify this in SimpleScreenRecorder, or maybe libvpx-vp9
I suggest that Vokoscreen had a menu to specify different options and an opportunity to manually edit ffmpeg options so that edited options had a priority over the ones specified using GUI menus

**Comment by vkohaupt** [comment]
Thanks for your suggestions,

The preset option comes of my todolist.
But the current preset is a very good mix of cpu-load, monitor-size, 
file-size, static- and animation picture.

A manually edit from ffmpeg options hmm.... no I do this not. A good 
functionality with the Gui is not guaranteed. If user want edit the 
ffmpeg options, can do this, copy it from vokoscreen and then can do 
what he wants.

Make matters worse it gives hundred versions of ffmpeg with different 
compiled options, and all distribution comes with other librarys. It is 
not easy to becomes a functional gui.

Have user problem with the packages from his distribution, is it a good 
idea to downlad my binary packages

http://linuxecke.volkoh.de/vokoscreen/vokoscreen.html

Volker

**Comment by Phr33d0m** [comment]
Hello @vkohaupt , the idea is to have an input field where Vokoscreen shows what parameters it will run with once you hit record, so you can edit them as you need. Currently I use Vokoscreen only to get ffmpeg command for the selected area, then I manually re-run it with different parameters.

For example it will run by default with:
```/usr/bin/ffmpeg -report -f x11grab -draw_mouse 1 -framerate 60 -video_size 1410x894 -i :0+0,94 -pix_fmt yuv420p -c:v libx264 -preset veryfast -qp 0 -q:v 1 -s 1410x894 -f matroska /tmp/vokoscreen-username/vokoscreen-2017-02-06_02-32-50.mkv```

But I need it to run with:
```-c:v h264_nvenc -profile:v high -pixel_format nv12 -preset default -rc vbr_2pass -rc-lookahead 32```

Currently I can do that by manually doing a lot of stuff but it will make my life way easier if the software allowed me to do this by default.

**TL;DR**: Request to implement a simple input with the ffmpeg run parameters before starting a recording where the user can modify at will.

**Comment by remussatala** [comment]
Very good idea.

**Comment by mikhailnov** [comment]
Yes, a great idea 06.02.2017, 04:36, "Phr33d0m" <notifications@github.com>:Hello @vkohaupt , the idea is to have an input field where Vokoscreen shows what parameters it will run with once you hit record, so you can edit them as you need. Currently I use Vokoscreen only to get ffmpeg command for the selected area, then I manually re-run it with different parameters.For example it will run by default with:/usr/bin/ffmpeg -report -f x11grab -draw_mouse 1 -framerate 60 -video_size 1410x894 -i :0+0,94 -pix_fmt yuv420p -c:v libx264 -preset veryfast -qp 0 -q:v 1 -s 1410x894 -f matroska /tmp/vokoscreen-username/vokoscreen-2017-02-06_02-32-50.mkvBut I need it to run with:-c:v h264_nvenc -profile:v high -pixel_format nv12 -preset default -rc vbr_2pass -rc-lookahead 32Currently I can do that by manually doing a lot of stuff but it will make my life way easier if the software allowed me to do this by default.TL;DR: Request to implement a simple input with the ffmpeg run parameters before starting a recording where the user can modify at will.—You are receiving this because you authored the thread.Reply to this email directly, view it on GitHub, or mute the thread.

**Comment by pjotrek-b** [comment]
I also think it'd be really great to be able to configure the encoding parameters somehow :smile:

**Comment by apienk** [comment]
+1 to the idea. If the parameters can not be fully editable I would only ask for lossless encoding of both video and audio, using for example `-pix_fmt yuv420p -c:v libx264 -preset veryfast -qp 0 -c:a pcm_s16le`.

**Comment by pjotrek-b** [comment]
@apienk: Good idea!
Yet, there are 2 things I'd like to mention about "lossless" encoding:

1) According to [ffmpeg docs for x264 encoding](https://trac.ffmpeg.org/wiki/Encode/H.264#LosslessH.264), lossless encoding requires the following parameter: `-crf 0`

2) For screen capture, I assume some RGB pix_fmt would be "more correct" for lossless? :wink: 
Just pointing out for the sake of completeness. :blush: (Don't even know if x264 does RGB though...)

**Comment by apienk** [comment]
1. No, `-qp 0` renders lossless content and is more strict than `-crf 0` [(please see this article)](http://slhck.info/video/2017/02/24/crf-guide.html).
2. Using RGB is a good idea on slow computers as the conversion to YUV takes up some processor time. Otherwise it's not recommended because x264-rgb is not supported on most platforms and players.
